lib/devise-security/hooks/session_limitable.rb

```diff
@@ -19,14 +24,13 @@
 # page on the next request.
 Warden::Manager.after_set_user only: :fetch do |record, warden, options|
   scope = options[:scope]
-  env   = warden.request.env
 
   if record.devise_modules.include?(:session_limitable) &&
      warden.authenticated?(scope) &&
      options[:store] != false
     if record.unique_session_id != warden.session(scope)['unique_session_id'] &&
-       !env['devise.skip_session_limitable'] &&
-       !record.skip_session_limitable?
+       !record.skip_session_limitable? && 
+       !warden.session(scope)['devise.skip_session_limitable']
```

Comment:
Prior to this change, we were able to modify the request by prepending `Devise::SessionController`'s `create` action, to trigger the  `skip_session_limitable` behavior:
```ruby
request.env['devise.skip_session_limitable'] = true
```

The (effective) change on this line no longer allows us this control:
```diff
-       !warden.request.env['devise.skip_session_limitable']
+       !warden.session(scope)['devise.skip_session_limitable']
```

Is there still a way to trigger the `skip_session_limitable` functionality in a similar manner?

Reply:
Hi @n00dle, I only just saw this comment. If this is still a problem for you, can you open up a new issue and we can discuss further there?

Reply:
Just a +1: I had the same problem here. With this change, preprended actions on my admin session controller do not work anymore to skip the limitable hook.

It's not really a problem since I can use the functionality as documented in the readme but I thought it would be interesting to document here that some devise hooks (like `trackable`) have that same functionality to skip the hook and implemented this by using the `warden.request.env` params + controller `prepended_before_action`. It seems weird to me that this gem uses the same hook definition but by using other mechanisms. In my case, I was using the `bypass_sign_in` method and since it does not accept keyword args I did not have a way to set the skip variable anymore.
If warden callbacks were a problem to me, it would result in a dead-end

Reply:
https://github.com/devise-security/devise-security/issues/421 please put any further discussion in here please! <3

Don't have the time to take a look into it right now unfortunately.